lib/core/api/get-file.js

```diff
@@ -24,43 +27,98 @@ module.exports = function(self) {
     throw e;
   };
 
+  function createFileToken(file, cb) {
+    file._client.createToken(file._bucketId, 'PULL', function(e, body) {
+      if(e) {
+        return _error(e, cb);
+      }
+
+      return handleTokens(file, body, cb);
+    });
+  }
+
+  function handleTokens(file, body, cb) {
+    if(body.encryptionKey === '' && file._key === undefined) {
+      return _error(
+        new Error('You must supply a keypair for private buckets'),
+        cb
+      );
+    }
+
+    file._encryptionKey = file._key;
+    if(body.encryptionKey !== undefined && body.encryptionKey !== '') {
+      file._encryptionKey = body.encryptionKey;
+    }
+
+    file._token = body.token;
+    file._id = body.id;
+    file._mimetype = body.mimetype;
+
+    getFilePointers(file, cb);
+  }
+
+  function getFilePointers(file, cb) {
+    var opts = {
+      token: file._token,
+      bucket: file._bucketId,
+      file: file._fileId
+    };
+
+    file._client.getFilePointers(opts, function(e, pointers) {
+      if(e) {
+        return _error(e);
+      }
+
+      resolveFileFromPointers(file, pointers, cb);
+    });
+  }
+
+  function resolveFileFromPointers(file, pointers, cb) {
+    file._client.resolveFileFromPointers(pointers, function(e, muxer) {
+      if(e) {
+        return _error(e);
+      }
+
+      file._muxer = muxer;
+      file.emit('ready');
+      saveFile(file);
+      return cb(null, file);
+    });
+  }
+
+  function saveFile(file) {
+    var stream = file._store.createWriteStream(file._fileId);
+    var key = KeyIV.getDeterministicKey(file._encryptionKey, file._fileId);
+    var secret = new KeyIV(key, file._fileId);
+    var decrypter = new DecryptStream(secret);
+    file._muxer.pipe(decrypter).pipe(stream);
+    file._muxer.on('error', file.emit.bind(file, 'error'));
+    stream.on('error', file.emit.bind(file, 'error'));
+    stream.on('data', file.emit.bind(file, 'data'));
+    file._muxer.on('end', function () {
+      stream.end();
+      delete file._muxer;
+      file.emit('done');
```

Comment:
Haven't tried yet though, might need to explicitly subclass readable stream. The thought process here is that I don't want a stream to try and manage back pressure (if there isn't a consumer, just discard the data emitted). But if you are listening to the `data` event, you can consume incomming data as quickly as it becomes available.

Also `renderTo`, `appendTo`, `getReadableStream` should all work as soon as the `File` object is ready, allowing you to `renderTo('#myvideo')` and the library will handle streaming to the rendered video player.